docs/src/artifacts.md

```diff
@@ -227,3 +227,65 @@ Note that while that hash was previously overridden, it is no longer, and theref
 Most methods that are affected by overrides have the ability to ignore overrides by setting `honor_overrides=false` as a keyword argument within them.
 For UUID/name based overrides to work, `Artifacts.toml` files must be loaded with the knowledge of the UUID of the loading package.
 This is deduced automatically by the `artifacts""` string macro, however if you are for some reason manually using the `Pkg.Artifacts` API within your package and you wish to honor overrides, you must provide the package UUID to API calls like `artifact_meta()` and `ensure_artifact_installed()` via the `pkg_uuid` keyword argument.
+
+## Extending Platform Selection
+
+!!! compat "Julia 1.6"
+    Pkg's extended platform selection requires at least Julia 1.6, and is considered experimental.
+
+New in Julia 1.6, `Platform` objects can have extended attributes applied to them, allowing artifacts to be tagged with things such as CUDA driver version compatibility, microarchitectural compatibility, julia version compatibility and more!
+Note that this feature is considered experimental and may change in the future.
+If you as a package developer find yourself needing this feature, please get in contact with us so it can evolve for the benefit of the whole ecosystem.
+In order to support artifact selection at `Pkg.add()` time, `Pkg` will run the specially-named file `<project_root>/.pkg/select_artifacts_hook.jl` and which should output a `TOML`-serialized dictionary representing the artifacts that this package needs.
+The format of the dictionary should match that returned from `Artifacts.select_downloadable_artifacts()`, and indeed most packages should simply call that function with an augmented `Platform` object.
+An example hook definition might look like the following, split across two files:
+
+```julia
+# .pkg/platform_augmentation.jl
+using Libdl, Base.BinaryPlatforms
```

Comment:
What packages are you allowed to load in this file? Why can you load `Libdl`?

Reply:
I assumed that it would be fine to load any standard libraries, but perhaps we should build an explicit list.  I think that `Artifacts`, `Libdl` and `TOML` are the bare minimum.

Reply:
That doesn't seem 100% viable to me. A package can only load its dependencies, stdlibs included. Alternatively, we can say that Pkg is the package that runs this file and then Pkg needs to take on a dependency on everything that is allowed to be loaded.

Reply:
It feels weird that it's the `Pkg` environment.  I'd rather ship a `Project.toml` within `Pkg` that has the explicit list embedded within it.

Reply:
Yeah, thinking back on it, I agree it is weird to use the Pkg one.